File: Anul 2/Semestrul 2/MPP/labs/Tema 5-proiect Server Csharp-Client Java/proiectSCcsharp/angajatModel/validators/SpectacolValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace proiectSCcsharp.domain.validators
{
    public class SpectacolValidator : Validator<Spectacol>
    {
        public SpectacolValidator()
        {
        }

        public void validate(Spectacol entity)
        {
            String errors = "";
            if (entity.artistName == "")
                errors += "invalid artist name\n";
            if (entity.place == "")
                errors += "invalid place\n";
            if (entity.availableSeats < 0)
                errors += "invalid available seats\n";
            if (entity.soldSeats< 0)
                errors += "invalid sold seats\n";
            if (errors.Length > 0)
                throw new ValidationException(errors);
        }
    }
}

File: Anul 2/Semestrul 2/MPP/labs/Tema 7-servicii REST/Java/Rest/proiect/Domain/src/main/java/proiect/domain/validators/SpectacolValidator.java
package proiect.domain.validators;

import proiect.domain.Spectacol;

import java.util.Objects;

public class SpectacolValidator implements Validator<Spectacol> {
    private static final SpectacolValidator instance = new SpectacolValidator();

    public static SpectacolValidator getInstance() {
        return instance;
    }

    public SpectacolValidator() {
    }

    @Override
    public void validate(Spectacol entity) throws ValidationException {
        String errors = "";
        if (Objects.equals(entity.getArtistName(), ""))
            errors += "invalid artist name\n";
        if (Objects.equals(entity.getDate(), ""))
            errors += "invalid date\n";
        if (Objects.equals(entity.getPlace(), ""))
            errors += "invalid place\n";
        if (entity.getAvailableSeats()<0)
            errors += "invalid available seats\n";
        if (entity.getSoldSeats()<0)
            errors += "invalid sold seats\n";
        if (errors.length() > 0)
            throw new ValidationException(errors);
    }
}
